test/sql/tdigest.sql

```diff
@@ -0,0 +1,139 @@
+
+--- Test t-digest I/O
+
+-- Test t-digest in/out
+SELECT (SELECT percentile_approx(_timescaledb_internal.tdigest_in(_timescaledb_internal.tdigest_out(tdigest_create(quantity))), .5) FROM test_table) 
+= (SELECT percentile_approx(tdigest_create(quantity), .5) FROM test_table) AS inout_works;
+
+-- Test t-digest send/recv
+\c :TEST_DBNAME :ROLE_SUPERUSER
+CREATE FUNCTION tdigest_test_sendrecv(tdigest)
+RETURNS boolean
+AS :MODULE_PATHNAME, 'ts_tdigest_test_sendrecv'
+LANGUAGE C STRICT IMMUTABLE PARALLEL SAFE;
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+
+SELECT (SELECT tdigest_test_sendrecv((SELECT tdigest_create(quantity) FROM test_table))) = true AS sendrecv_works;
+
+--- Test Percentile 
+--- tdigest is not deterministically generated due to parallelization changing merge orders, so we only check accuracy
+
+-- Test Percentile Accuracy -- currently, tests for percent error between exact and estimated of no more than 1%
+\set APPROX_FUNC 'percentile_approx'
+\set EXACT_FUNC 'percentile_cont'
+\set ACCURACY_REQUIREMENT '.01'
+\set TEST_COLUMN 'quantity'
+\set TEST_TABLE 'test_table'
+
+\set TEST_PARAM '0'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.1'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.2'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.3'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.4'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.5'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.6'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.7'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.8'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '0.9'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '1'
+:CALCULATE_PERCENT_ERROR
+
+-- Make sure we can't call with invalid percentile values
+\set ON_ERROR_STOP 0
+SELECT percentile_approx(tdigest_create(quantity), 2) FROM test_table;
+SELECT percentile_approx(tdigest_create(quantity), -1) FROM test_table;
+\set ON_ERROR_STOP 1
+
+--- Test CDF 
+
+-- Test CDF Accuracy 
+\set APPROX_FUNC 'cdf_approx'
+\set EXACT_FUNC 'cume_dist'
+\set ACCURACY_REQUIREMENT '.01'
+\set TEST_COLUMN 'quantity'
+\set TEST_TABLE 'test_table'
+
+\set TEST_PARAM '63'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '42'
+:CALCULATE_PERCENT_ERROR
+
+\set TEST_PARAM '100'
+:CALCULATE_PERCENT_ERROR
+
+--- Set timestamp for continuous aggs
+SET timescaledb.current_timestamp_mock = '2018-01-01 00:00';
+
+--- Test continuous agg creation
+CREATE OR REPLACE VIEW cagg_view WITH (timescaledb.continuous, timescaledb.refresh_lag = -1, timescaledb.max_interval_per_job = '10000 days') AS
+SELECT time_bucket('1w', time) AS week, tdigest_create(quantity) AS tdigest
+FROM test_table
+GROUP BY time_bucket('1w', time);
+
+REFRESH MATERIALIZED VIEW cagg_view;
+
+--- Test querying from cont agg/view/table
+SELECT count(percentile_approx(tdigest, .5)) FROM cagg_view;
+SELECT count(cdf_approx(tdigest, 42)) FROM cagg_view;
+SELECT cdf_approx(tdigest, 42) FROM cagg_view LIMIT 5 ;
+SELECT percentile_approx(tdigest, .5) FROM cagg_view LIMIT 5;
+
```

Comment:
need tests that verify t-digest is being updated correctly.
for each of these cases: inserts/updates and deletes to "test_table"  .
Then Refresh the cont. agg view and verify the tdigest gets updated by querying cont agg.